taichi/backends/vulkan/codegen_vulkan.cpp

```diff
@@ -49,6 +49,7 @@ class TaskCodegen : public IRVisitor {
  public:
   struct Params {
     OffloadedStmt *task_ir;
+    VkRuntime *runtime;
```

Comment:
Is it possible to pass in just the `VulkanCapabilities`? The reason is that we should have a clean separation between the runtime stuff and codegen stuff. (In the future I'd like to completely separate them into different folders, e.g. `runtime/` and `codegen/`

Reply:
Or at least leave a TODO here:

```suggestion
    // TODO: Pass in a VulkanCapabilities directly.
    VkRuntime *runtime;
```